File: Backend/erpc/src/main/java/com/cineverse/erpc/admin/delete/service/DeleteService.java
package com.cineverse.erpc.admin.delete.service;

import com.cineverse.erpc.admin.delete.dto.account.RequestAccountDeleteRequestProcess;
import com.cineverse.erpc.admin.delete.dto.account.ResponseAccountDeleteRequestList;
import com.cineverse.erpc.admin.delete.dto.account.ResponseAccountDeleteRequestProcess;
import com.cineverse.erpc.admin.delete.dto.account.ResponseFindAccoundDeleteRequest;
import com.cineverse.erpc.admin.delete.dto.order.RequestOrderDeleteRequestProcess;
import com.cineverse.erpc.admin.delete.dto.order.ResponseFindOrderDeleteRequest;
import com.cineverse.erpc.admin.delete.dto.order.ResponseOrderDeleteRequestList;
import com.cineverse.erpc.admin.delete.dto.order.ResponseOrderDeleteRequestProcess;
import com.cineverse.erpc.admin.delete.dto.quotation.RequestQuotationDeleteRequestProcess;
import com.cineverse.erpc.admin.delete.dto.quotation.ResponseFindQuotationDeleteRequest;
import com.cineverse.erpc.admin.delete.dto.quotation.ResponseQuotationDeleteRequestList;
import com.cineverse.erpc.admin.delete.dto.quotation.ResponseQuotationDeleteRequestProcess;
import com.cineverse.erpc.contract.aggregate.ContractDeleteRequest;
import com.cineverse.erpc.contract.dto.ContractDeleteRequestDTO;
import com.cineverse.erpc.order.order.dto.ResponseDeleteOrder;
import com.cineverse.erpc.salesopp.opportunity.aggregate.SalesOppDeleteRequest;
import com.cineverse.erpc.salesopp.opportunity.dto.SalesOppDeleteRequestDTO;

import java.util.List;

public interface DeleteService {

    List<SalesOppDeleteRequest> findSalesOppDeleteRequestList();

    SalesOppDeleteRequestDTO findSalesOppDeleteRequestById(long salesOppDeleteRequestId);

    SalesOppDeleteRequest changeOppDeleteRequestStatus(long salesOppDeleteRequestId,
                                                       SalesOppDeleteRequestDTO deleteOpp);

    List<ContractDeleteRequest> findContractDeleteRequestList();

    ContractDeleteRequestDTO findContractDeleteRequestById(long contractDeleteRequestId);

    ContractDeleteRequest changeContractDeleteRequestStatus(ContractDeleteRequestDTO deleteContractDTO,
                                                            long contractDeleteRequestId);

    List<ResponseQuotationDeleteRequestList> findQuotationDeleteRequestList();

    ResponseFindQuotationDeleteRequest findQuotationDeleteRequestById(long quotationDeleteRequestId);

    ResponseQuotationDeleteRequestProcess processQuotationDeleteRequest(RequestQuotationDeleteRequestProcess requestQuotationDeleteRequestProcess);

    List<ResponseAccountDeleteRequestList> findAccountDeleteList();

    ResponseFindAccoundDeleteRequest findAccountDeleteRequestById(long accountDeleteRequestId);

    ResponseAccountDeleteRequestProcess accountDeleteRequestProcess(RequestAccountDeleteRequestProcess requestAccountDeleteRequestProcess);

    List<ResponseOrderDeleteRequestList> findOrderDeleteRequestList();

    ResponseFindOrderDeleteRequest findOrderDeleteRequestById(long orderDeleteRequestId);

    ResponseOrderDeleteRequestProcess processOrderDeleteRequest(RequestOrderDeleteRequestProcess requestOrderDeleteRequestProcess);
}

File: erpc-front/src/components/delete/DeleteService.js
import axios from 'axios';

const API_URL = 'http://erpc-back-ver2-env.eba-3inzi7ji.ap-northeast-2.elasticbeanstalk.com/delete';

class DeleteService {
  constructor() {
    this.api = axios.create({
      baseURL: API_URL,
    });
  }


  getSalesOppDeleteRequests() {
    return this.api.get('/sales_opp');
  }

  findSalesOppDeleteRequestById(salesOppDeleteRequestId) {
    return this.api.get(`/sales_opp/${salesOppDeleteRequestId}`);
  }

  updateSalesOppDeleteRequestProcess(data) {
    return this.api.patch('/sales_opp/process', data);
  }

  // Contract Delete Requests
  getContractDeleteRequests() {
    return this.api.get('/contract');
  }

  findContractDeleteRequestById(contractDeleteRequestId) {
    return this.api.get(`/contract/${contractDeleteRequestId}`);
  }

  updateContractDeleteRequestStatus( data) {
    return this.api.patch('/contract/process', data);
  }

  // Quotation Delete Requests
  getQuotationDeleteRequests() {
    return this.api.get('/quotation');
  }

  findQuotationDeleteRequestById(quotationDeleteRequestId) {
    return this.api.get(`/quotation/${quotationDeleteRequestId}`);
  }

  processQuotationDeleteRequest( data) {
    return this.api.patch('/quotation/permission', data);
  }

  // Account Delete Requests
  getAccountDeleteRequests() {
    return this.api.get('/account');
  }

  findAccountDeleteRequestById(accountDeleteRequestId) {
    return this.api.get(`/account/${accountDeleteRequestId}`);
  }

  processAccountDeleteRequest(data) {
    return this.api.patch('/account/process', data);
  }


  getOrderDeleteRequests() {
    return this.api.get('/order');
  }

  findOrderDeleteRequestById(orderDeleteRequestsId) {
    return this.api.get(`/order/${orderDeleteRequestsId}`);
  }

  processOrderDeleteRequest(data) {
    return this.api.patch('/order/process', data);
  }
}

export default new DeleteService();